### DISCUSSION TITLE
Discussion on creating Basis files

### DISCUSSION THREAD

**Discussion by tsoen** [atteneder/KtxUnity#61] [Q&A]
Hello @atteneder,

Kind of a silly question here, but since you've worked with .basis files, have you or anyone ever heard of a way to create .basis files on Android, during runtime... since the **basisu** command line tool cannot be executed on a Android device.
I would like to be able to pick a file on the Android device and compress it on the fly.

Any idea or workaround ?

Regards,

**Comment by atteneder** [comment]
Hi @tsoen ,

Not a silly question at all.

I'm not aware of a Unity-integrated implementation nor any Android port of basisu encoder/libktx, so somebody⟨™⟩ has to do it.

At some point I'll tackle encoding to KTX, but I have no plans on the .basis container format.

BasisU encoding is quite demanding, computationally. It'll take some time and drain the battery if done at scale. Consider other encoders (jpeg? would be available already) and a server-side conversion to basis.

hth
